cmake/install/Panda3DConfig.cmake

```diff
@@ -0,0 +1,153 @@
+# PANDA 3D SOFTWARE
+# Copyright (c) Carnegie Mellon University.  All rights reserved.
+#
+# All use of this software is subject to the terms of the revised BSD
+# license.  You should have received a copy of this license along
+# with this source code in a file named "LICENSE."
+#
+# Author: CFSworks (Dec. 11, 2018)
+
+# This file is installed to CMake's package search path, and is invoked for
+# find_package(Panda3D [COMPONENTS ...])
+#
+# The following components are available for importing:
+#
+#   Core      - The core Panda3D libraries; this component is always included.
+#
+#               Panda3D::Core::panda
+#               Panda3D::Core::pandaexpress
+#               etc.
+#
+#
+#   Python    - Python targets, which can be used for linking against the Python
+#               extension modules directly.  Note that this also imports the
+#               Python bindings for other requested components that have them.
+#
+#               Panda3D::Python::panda3d.core
+#               Panda3D::Python::panda3d.physics
+#               etc.
+#
+#
+#   Tools     - Various tools used in asset manipulation and debugging.
+#
+#               Panda3D::Tools::egg2bam
+#               Panda3D::Tools::egg-optchar
+#               Panda3D::Tools::pview
+#               etc.
+#
+#
+#   Direct    - Panda's "direct" Python framework; C++ support library.
+#
+#               Panda3D::Direct::p3direct
+#
+#
+#   Contrib   - Extensions not part of the Panda3D core, but contributed by the
+#               community.
+#
+#               Panda3D::Contrib::p3ai
+#               Panda3D::Contrib::p3rplight
+#
+#
+#   Framework - Panda's "p3framework" C++ framework.
+#
+#               Panda3D::Framework::p3framework
+#
+#
+#   Egg       - Support for the Egg file format.
+#
+#               Panda3D::Egg::pandaegg
+#
+#
+#   Bullet    - Support for Bullet physics.
+#
+#               Panda3D::Bullet::p3bullet
+#
+#
+#   ODE       - Support for the ODE physics engine.
+#
+#               Panda3D::ODE::p3ode
+#
+#
+#   FFmpeg    - Support for FFmpeg media format loading.
+#
+#               Panda3D::FFmpeg::p3ffmpeg
+#
+#
+#   OpenAL    - Support for OpenAL audio output.
+#
+#               Panda3D::OpenAL::p3openal_audio
+#
+#
+#   FMOD      - Support for FMOD audio output.
+#
+#               Panda3D::OpenAL::p3fmod_audio
+#
+#
+#   OpenGL    - Support for OpenGL rendering.
+#
+#               Panda3D::OpenGL::pandagl
+#
+#
+#   DX9       - Support for Direct3D 9 rendering.
+#
+#               Panda3D::DX9::pandadx9
+#
+#
+#   GLES      - Support for OpenGL ES rendering.
+#
+#               Panda3D::GLES::pandagles
+#               Panda3D::GLES::pandagles2
```

Comment:
Components should correspond mostly 1:1 to checkboxes in the NSIS installer. I could see someone wanting desktop OpenGL but not GLES, so it makes sense not to put them under `OpenGL`. However you may have a good point that they should be split up - I grouped them because they're so heavily grouped in the codebase.

Actually, perhaps it's best if you take a crack at the components list from a mile up: if these are options for what can/can't be installed independently, do they make sense? What would be your ideal component listing (off the top of your head, without even looking at code)?

Reply:
Off the top of my head?  One for every `panda3d.*` module, and then one for every plug-in.  So, `Core`, `Direct`, `Vision`, `Assimp`, `FFMpeg`, `OpenGL`, `OpenGLES2`, `OpenAL`, etc.

Upon further thought I think I could go either way on splitting up GLES.  They're somewhat related in that a target platform probably supports both.  But, one will probably choose one or the other.  (I could actually see us deprecating OpenGL ES 1 at some point in the future.)